### PR TITLE
KAZOO-4029: fix CNAM -> CNAME for DNS, support SRV results to binary …

### PR DESCRIPTION
**Pull request by k-anderson**
…and preform lookups on the nameserver for the domain.

### CODE REVIEW

**Comment by k-anderson** [comment]
I need to test NAPTR before we merge this but I wanted to start the review early ;)

File: applications/crossbar/src/modules/cb_whitelabel.erl [hunk 1]
```diff
@@ -507,53 +506,67 @@ test_account_domains(Context) ->
...
--spec test_host_mappings(ne_binary(), ne_binaries(), ne_binary()) ->
-                                wh_json:object().
-test_host_mappings(Host, Mappings, <<"CNAM">>) ->
-    %% inet_dns wants it with an e
-    test_host_mappings(Host, Mappings, <<"CNAME">>);
-test_host_mappings(Host, Mappings, DomainType) ->
-    wh_json:from_list([{<<"expected">>, Mappings}
-                       ,{<<"actual">>, lookup(Host, DomainType)}
+test_host(Host, HostConfig, DomainType, Options) ->
+    io:format("host: ~p~nhost config: ~p~ntype: ~p~n", [Host, HostConfig, DomainType]),
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_whitelabel.erl]: Did you want this still? Convert to lager if so...

File: core/whistle-1.0.0/src/wh_network_utils.erl [hunk 2]
```diff
@@ -207,15 +335,101 @@ pretty_print_bytes(Bytes) ->
...
+    ].
+
+-spec set_option_usevc(options(), boolean()) -> options().
+set_option_usevc(Options, Value) ->
+    [{'usevc', Value}
+     |props:delete('usevc', Options)
+    ].
+
+
+maybe_resolve_nameservers([], Nameservers) -> Nameservers;
```
Comment (jamesaimonetti) [inline core/whistle-1.0.0/src/wh_network_utils.erl]: Spec

File: core/whistle-1.0.0/src/wh_network_utils.erl [hunk 3]
```diff
@@ -207,15 +335,101 @@ pretty_print_bytes(Bytes) ->
...
+
+-spec default_options() -> options().
+default_options() -> ?LOOKUP_OPTIONS.
+
+-type nameserver() :: {inet:ip_address(), Port :: 1..65535}.
+
+-spec set_option_alt_nameservers(options(), [nameserver()]) -> options().
+set_option_alt_nameservers(Options, Value) ->
+    [{'alt_nameservers', maybe_resolve_nameservers(Value, [])}
+     |props:delete('alt_nameservers', Options)
```
Comment (jamesaimonetti) [inline core/whistle-1.0.0/src/wh_network_utils.erl]: `props:set_value/3` will accomplish this

**Comment by k-anderson** [comment]
tested all record types and they seem to work now!

**Comment by k-anderson** [comment]
Seems @jamesaimonetti accepted this when the check failed, and fixed it on master!  I also pushed a fix for the schema so they conflict.  Since the check now passed and its on master I will close the PR :)

**Pull request closed without merging**